packages/react-core/src/components/Dropdown/DropdownMenu.tsx

```diff
@@ -152,11 +152,25 @@ export class DropdownMenu extends React.Component<DropdownMenuProps> {
         return React.cloneElement(group, props);
       });
     }
-    return React.Children.map(children, (child, index) =>
-      React.cloneElement(child as React.ReactElement, {
-        index
-      })
-    );
+    return React.Children.map(children, (child, index) => this.recursiveMap(child, index));
+  }
+
+  recursiveMap(children: any, index: number) {
+    return React.Children.map(children, child => {
+      if (!React.isValidElement(child)) {
+        return child;
+      }
+
+      if ((child.props as any).children) {
+        const obj = {
+          children: this.recursiveMap((child.props as any).children, index),
+          index
```

Comment:
Should we do something like:

```suggestion
          index: child.props.index ?? index
```

So that we don't overwrite an existing index prop?

Reply:
Also, if downstream user uses their own `index` prop on a Dropdown item or any of its children, would this cause keyboard nav to misbehave?